MohammadBallal/main.py

```diff
@@ -0,0 +1,117 @@
+from fastapi import FastAPI, HTTPException, Path
+from pydantic import BaseModel
+from uuid import UUID, uuid4
+from datetime import date
+from enum import Enum
+
+
+
+app = FastAPI()
+
+# Using enum for status and priority
+class Status(Enum):
+    InProgress = "InProgress"
+    Completed = "Completed"
+
+class Priority(Enum):
+    High = "High"
+    Medium = "Medium"
+    Low = "Low"
+
+
+# Task model
+class Task(BaseModel):
+    task_id: UUID= uuid4()  # Automatically generates a unique ID for each task
+    title: str                                
+    description: str = "No description provided"                           
+    due_date: date                            
+    status: Status 
+    priority: Priority = None
+
+    
+tasks= []
```

Comment:
In my opinion, using `dict` is better and easier that using two lists

Reply:
I understand you, and if this is necessary I will make the adjustment if you want.